### ISSUE TITLE
losses and results

### ISSUE THREAD

**Issue by Wenchao-Du** [zzangjinsun/NLSPN_ECCV20#27]
Hi, this is a interesting work for depth estimation, i have some problems in your project:
1. losses from final prediction, why do not add a loss for initial prediction in your training instead of only final supervision, and have you tried it in trianing?  
2. i have tried the released model for kitti dataset, but i got the wrong results that RMSE & MAE > 10000(mm), my env is python=3.6 torch=1.6 torchvision=0.7 on ubuntu 16.04, thanks

**Comment by zzangjinsun** [comment]
1. I tried supervision on the initial depth but I thought it was too strong so refinement from the propagation was not good enough.

2. Are you using the same arguments for testing?

python main.py --dir_data PATH_TO_KITTI_DC --data_name KITTIDC --split_json ../data_json/kitti_dc.json \
    --patch_height 240 --patch_width 1216 --gpus 0,1,2,3 --max_depth 90.0 --num_sample 0 \
    --test_only --pretrain ../results/NLSPN_KITTI_DC.pt --preserve_input --save NAME_TO_SAVE --legacy

You might miss some arguments such as --num_sample 0 or --legacy

**Comment by Wenchao-Du** [comment]
Thanks for your reply, the initial prediction brings more uncertainty if without supervision. If different weights for initial and final predictions are tried by you? For testing, I ignored the legency param and tested on the single GPU.

**Comment by zzangjinsun** [comment]
I haven't tried weighting between the initial and the final predictions.

For the pretrained models, --legacy flag is essential. Please try with that option.

**Comment by Wenchao-Du** [comment]
Thanks, i will try it
